Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPISidebar.mm

```diff
@@ -0,0 +1,243 @@
+    @"background": @{
+        @"scripts": @[ @"background.js", ],
+        @"type": @"module",
+        @"persistent": @NO,
+    },
+
+    @"sidebar_action": @{
+        @"default_title": @"Test Sidebar",
+        @"default_panel": @"sidebar.html",
+        @"default_icon": @{
+            @"16": @"sidebar-16.png",
+            @"32": @"sidebar-32.png",
+        },
+        @"open_at_install": @NO,
+    },
+};
+
+
+static auto *sidePanelManifest = @{
+    @"manifest_version": @3,
+
+    @"name": @"SidePanel Test",
+    @"description": @"SidePanel Test",
+    @"version": @"1",
+
+    @"permissions": @[ @"sidePanel" ],
+    @"background": @{
+        @"scripts": @[ @"background.js" ],
+        @"type": @"module",
+        @"persistent": @NO,
+    },
+};
+
+static auto *sidebarActionAndPanelManifest = @{
+    @"manifest_version": @3,
+
+    @"name": @"Sidebar Test (both)",
+    @"description": @"Sidebar Test (both)",
+    @"version": @"1",
+
+    @"permissions": @[ @"sidePanel" ],
+
+    @"background": @{
+        @"scripts": @[ @"background.js" ],
+        @"type": @"module",
+        @"persistent": @NO,
+    },
+
+    @"sidebar_action": @{
+        @"default_title": @"Test Sidebar",
+        @"default_panel": @"sidebar.html",
+        @"default_icon": @{
+            @"16": @"sidebar-16.png",
+            @"32": @"sidebar-32.png",
+        },
+        @"open_at_install": @NO,
+    },
+};
+
+static auto *neitherSidebarActionNorPanelManifest = @{
+    @"manifest_version": @3,
+    @"name": @"Sidebar Test (neither)",
+    @"description": @"Sidebar Test (neither)",
+    @"version": @"1",
+
+    @"permissions": @[],
+    @"background": @{
+        @"scripts": @[ @"background.js", ],
+        @"type": @"module",
+        @"persistent": @NO,
+    },
+};
+
+// This test fixture allows us to use sidebarConfig (which enables the sidebar feature flag) without manually constructing one on each run
+class WKWebExtensionAPISidebarTest : public testing::Test {
+protected:
+    WKWebExtensionAPISidebarTest()
+    {
+        sidebarConfig = _WKWebExtensionControllerConfiguration.defaultConfiguration;
+        if (!sidebarConfig.webViewConfiguration)
+            sidebarConfig.webViewConfiguration = [[WKWebViewConfiguration alloc] init];
+
+        for (_WKFeature *feature in [WKPreferences _features]) {
+            if ([feature.key isEqualToString:@"WebExtensionSidebarEnabled"])
+                [sidebarConfig.webViewConfiguration.preferences _setEnabled:YES forFeature:feature];
+        }
+    }
+
+    _WKWebExtensionControllerConfiguration *sidebarConfig;
+};
+
+TEST_F(WKWebExtensionAPISidebarTest, APISUnavailableWhenManifestDoesNotRequest)
+{
+    auto *script = @[
+        @"browser.test.assertDeepEq(browser.sidebarAction, undefined)",
+        @"browser.test.assertDeepEq(browser.sidePanel, undefined)",
+        @"browser.test.assertDeepEq(chrome.sidebarAction, undefined)",
+        @"browser.test.assertDeepEq(chrome.sidePanel, undefined)",
```

Comment:
Nit: I'm not sure we need the tests of the `chrome` namespace. We alias it to browser everywhere and I believe that is tested elsewhere.